Model/Session.php

```diff
@@ -6,8 +6,24 @@
 
 /**
  * Message session model
+ *
+ * @method bool hasProductsAddedToCart() Magic method, true if "products_added_to_cart" is set
+ * @method mixed getProductsAddedToCart() Magic method, returns array of unclaimed "products_added_to_cart" events or null if none
  */
 class Session extends SessionManager
 {
+    /**
+     * Saves "product_added_to_cart" events in session
+     *
+     * @param array $events
+     * @return void
+     */
+    public function addProductsAddedToCart(array $events): void
+    {
+        $sessionData = $this->getProductsAddedToCart() ?: [];
 
+        array_push($sessionData, ...$events);
```

Comment:
What happens if a customer deletes some products from their cart and then add other products?
`products_added_to_cart` will contain a history of all of the products that were added to cart, is that the needed logic?

**UPDATE:**
Would the `checkout/cart/add` action delete the cache in the internal storage? So the array would be reconfigured when adding products to the cart?